components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -150,13 +175,93 @@ function AutomaticPointsTable:queryPlacements(tournaments)
 			local tournamentIndex = tournamentIndices[result.tournament]
 			local tournament = tournaments[tournamentIndex]
 
-			result.points = tonumber(result.extradata.prizepoints)
+			result.prizePoints = tonumber(result.extradata.prizepoints)
 			result.securedPoints = tonumber(result.extradata.securedpoints)
+			result.extradata = nil
 			table.insert(tournament.placements, result)
+
+			local participant = result.participant
+			local teamIndex = reverseAliasLookupTable[tournamentIndex][participant]
+			if teamIndex ~= nil then
+				teams[teamIndex].results[tournamentIndex] = result
+			end
 		end
 	)
 
-	return tournaments
+	return teams, tournaments
+end
+
+function AutomaticPointsTable:getPointsData(teams, tournaments)
+	local pointsData = {}
+	Table.iter.forEachIndexed(teams,
+		function(teamIndex, team)
+			local teamPointsData = {}
+			local totalPoints = 0
+			for tournamentIndex = 1, #tournaments do
+				local manualPoints = team.manualPoints[tournamentIndex]
+				local placement = team.results[tournamentIndex]
+				local tournamentTeamPointsData = self:getTournamentTeamPointsData(manualPoints, placement)
+				if Table.isNotEmpty(tournamentTeamPointsData) then
+					totalPoints = totalPoints + tournamentTeamPointsData.amount
+				end
+
+				local deduction = team.deductions[tournamentIndex]
+				local dataWithDeduction = self:getDeductionData(deduction, tournamentTeamPointsData)

```

Comment:
Use some variation of the below and scrap the function

```suggestion
				pointsForTournament.deduction = (deduction ~= nil and deduction.amount ~= nil and deduction.amount) or 0
```

Can even remove the condition below.